### DISCUSSION TITLE
getting different results with api vs javascript sdk

### DISCUSSION THREAD

**Discussion by TreestanG** [pocketbase/pocketbase#5509] [Q&A]
I seem to be getting different results with my js sdk compared to directly accessing the url in my browser. 
![image](https://github.com/user-attachments/assets/75a3b3e3-ce90-4cb8-a151-3be955727222)
Currently, I have 2 records in my `courses` collection and if i access http://127.0.0.1:8090/api/collections/courses/records, it gives me this:
```
{"page":1,"perPage":30,"totalItems":2,"totalPages":1,"items":[{"author":"hs9pxtakhxfxiec","collectionId":"7zdzlft1eaqn7cm","collectionName":"courses","created":"2024-09-06 05:02:21.106Z","description":"learn more about python in this course. basics to python.","id":"09264ej49luxfnd","name":"Intro to Python","students":[],"updated":"2024-09-06 05:02:21.106Z"},{"author":"hs9pxtakhxfxiec","collectionId":"7zdzlft1eaqn7cm","collectionName":"courses","created":"2024-09-11 02:44:27.425Z","description":"learn more about python. has intermediate python","id":"1fmivzqj5g44zi4","name":"Intermediate Python","students":[],"updated":"2024-09-11 02:44:27.425Z"}]}
```
However, if I try and access the same path with js sdk:
```js
    console.log(await pb.collection('courses').getFullList())
```
It logs this:
```
[
  {
    collectionId: '7zdzlft1eaqn7cm',
    collectionName: 'courses',
    created: '2024-09-06 05:02:21.106Z',
    created_by: 'hs9pxtakhxfxiec',
    description: 'learn more about python in this course. basics to python.',
    id: '09264ej49luxfnd',
    name: 'Intro to Python',
    students: [],
    updated: '2024-09-06 05:02:21.106Z'
  }
]
```
I'm looking at my logs in pocketbase, as I'm running it through dev mode, and it says that there's a limit of 1 for some reason. I looked at the docs and it said that the `getFullList` method should return all records inside a collection so I'm not too sure why this is happening.

Thanks in advance!

**Comment by ganigeorgiev** [comment]
Check your collection List/Search API rule and make sure that your client request satisfies it.

`getList`, `getFullList`, and any other request method returns only the records the user has access to (aka. API rules act as filters and in the Admin UI you can see all records because admins have access to everything).

**Reply by ganigeorgiev** [reply]
Also if you are using Next.js, make sure that what you are experiencing is not caused by its [default cache](https://nextjs.org/docs/app/building-your-application/caching).

**Reply by TreestanG** [reply]
Ah I see, thanks! I'll look into it. I am using nextjs so this is most likely the issue. My list/search rule is empty meaning that everyone should have access.
